src/redux/reducers/authReducer.js

```diff
@@ -1,10 +1,42 @@
-// import * as types from '../types';
+import { LOGIN_SUCCESS, LOGIN_FAIL, LOGOUT, SET_LOADING } from '../types';
 
-const defaultState = {};
+const defaultState = {
+  token: null,
+  isAuthenticated: null,
+  loading: false,
+  user: null,
+  error: null,
+};
 
 export default (state = defaultState, action) => {
   switch (action.type) {
+    case LOGIN_SUCCESS:
+      if (action.payload.message) {
+        return state;
+      }
```

Comment:
what is this `if` condition actually doing?

Reply:
This prevents the ```isAuthenticated``` to be set to``` true``` after a user closes the popup without logging in. Since the backend sends a message if a token is not generated, I have used that as a check.

Reply:
Okay